applicationset/controllers/applicationset_controller.go

```diff
@@ -560,6 +581,7 @@ func (r *ApplicationSetReconciler) SetupWithManager(mgr ctrl.Manager, enableProg
 	}).For(&argov1alpha1.ApplicationSet{}).
```

Comment:
```suggestion
For(&argov1alpha1.ApplicationSet{}, builder.WithPredicates(ignoreWhenAnnotationApplicationSetRefreshIsRemoved())).
```
`WithEventFilter` will run for all objects that trigger a reconciliation. In this case, it would be `applications` and `applicationsets`. I suggest we move the predicate for applicationsets unless there is a good reason not to.

Reply:
You were right, using WithEventFilter was a bad idea, Moving predicate for appset is enough.